src/Manipulators/Size.php

```diff
@@ -271,102 +270,93 @@ public function runResize(Image $image, $fit, $width, $height)
+    public function runFillResize(ImageInterface $image, $width, $height): ImageInterface
     {
         $image = $this->runMaxResize($image, $width, $height);
 
-        return $image->resizeCanvas($width, $height, 'center');
+        return $image->resizeCanvas($width, $height, 'ffffff', 'center');
     }
 
     /**
      * Perform fill-max resize image manipulation.
      *
-     * @param Image $image  The source image.
-     * @param int   $width  The width.
-     * @param int   $height The height.
+     * @param ImageInterface $image  The source image.
+     * @param int            $width  The width.
+     * @param int            $height The height.
      *
-     * @return Image The manipulated image.
+     * @return ImageInterface The manipulated image.
      */
-    public function runFillMaxResize(Image $image, $width, $height)
+    public function runFillMaxResize(ImageInterface $image, $width, $height): ImageInterface
     {
-        $image = $image->resize($width, $height, function ($constraint) {
-            $constraint->aspectRatio();
-        });
+        $image = $image->resize($width, $height);
 
-        return $image->resizeCanvas($width, $height, 'center');
+        return $image->resizeCanvas($width, $height, 'ffffff', 'center');
     }
 
     /**
      * Perform stretch resize image manipulation.
      *
-     * @param Image $image  The source image.
-     * @param int   $width  The width.
-     * @param int   $height The height.
+     * @param ImageInterface $image  The source image.
+     * @param int            $width  The width.
+     * @param int            $height The height.
      *
-     * @return Image The manipulated image.
+     * @return ImageInterface The manipulated image.
      */
-    public function runStretchResize(Image $image, $width, $height)
+    public function runStretchResize(ImageInterface $image, $width, $height): ImageInterface
     {
         return $image->resize($width, $height);
     }
 
     /**
      * Perform crop resize image manipulation.
      *
-     * @param Image $image  The source image.
-     * @param int   $width  The width.
-     * @param int   $height The height.
+     * @param ImageInterface $image  The source image.
+     * @param int            $width  The width.
+     * @param int            $height The height.
      *
-     * @return Image The manipulated image.
+     * @return ImageInterface The manipulated image.
      */
-    public function runCropResize(Image $image, $width, $height)
+    public function runCropResize(ImageInterface $image, $width, $height): ImageInterface
     {
         list($resize_width, $resize_height) = $this->resolveCropResizeDimensions($image, $width, $height);
 
         $zoom = $this->getCrop()[2];
 
-        $image->resize($resize_width * $zoom, $resize_height * $zoom, function ($constraint) {
-            $constraint->aspectRatio();
-        });
+        $image->resize($resize_width * $zoom, $resize_height * $zoom);
```

Comment:
```suggestion
        $image->scale($resize_width * $zoom, $resize_height * $zoom);
```

We could probably also do away with the whole crop dimensions calculation by using the new [Image::cover()](https://image.intervention.io/v3/modifying/resizing#cropping--resizing-combined)

Reply:
I cannot find differences between v2 and v3 using query `?h=400&w=400&size=crop`.

![grafik](https://github.com/thephpleague/glide/assets/2162994/c0d1e856-34e3-45c5-98ad-84e76b28c69f)

Reply:
I don't see any cropping in either :)

Reply:
You are using incorrect query string param, you need to use `fit` instead of `size`.

Reply:
:facepalm: :sweat_smile: